pandas/core/window/rolling.py

```diff
@@ -2209,6 +2209,20 @@ def _apply(
         grouped_index_name = [*grouped_object_index.names]
         groupby_keys = [grouping.name for grouping in self._groupby.grouper._groupings]
         result_index_names = groupby_keys + grouped_index_name
+        drop_levels = []
+        obj = self._groupby._selected_obj
```

Comment:
this is way to special cases; we need to figure out *why* this is happening and adjust the logic more generally.

Reply:
This is happening because of 

```
        for key, values in self._groupby.grouper.indices.items():
            for value in values:
                data = [
                    *com.maybe_make_list(key),
                    *com.maybe_make_list(grouped_object_index[value]),
                ]
                result_index_data.append(tuple(data))

        result_index = MultiIndex.from_tuples(
            result_index_data, names=result_index_names
        )
```
which is a few lines below. ``self._groupby.grouper.indices.items()``represents the grouping columns and ``grouped_object_index[value]`` the index used for the rolling operation. If parts of the grouping columns are from the index, we duplicate these here. As I wrote, I could not figure out a way to avoid this within the loop, which would not be pretty complicated.